cpu/nrf52/Makefile.dep

```diff
@@ -17,5 +17,9 @@ ifneq (,$(filter nrf52832xxaa,$(CPU_MODEL)))
   endif
 endif
 
+ifneq (,$(filter saul_nrf_vddh,$(USEMODULE)))
+  FEATURES_REQUIRED += periph_adc
```

Comment:
Thinking about it, would it be cleaner to also depend this on the presence of the VDDH pin in general somehow? E.g. we would define a VDDH feature for the CPU families that do support it and require that feature here. That feature would be decoupled from the LC_FITLER_REG0 feature, as the VDDH voltage level is always present, independent if voltage regulator stage 0 is used or not.

Would that make sense?

Reply:
Right now the feature is only selected on a per board basis, as VDDH is not always uses even if it would be possible.